tests/unit/service_tests.py

```diff
@@ -66,3 +66,146 @@ def test01_get_vm(self):
                                       self.template_name,
                                       self.auth_token),
                        self.hostname)
+
+  @skipIf(SKIP_EVERYTHING, 'Skip if we are creating/modifying tests!')
+  def test02_get_vm_neg(self):
+    """Negative test case for attempting to retrieve non-existent VM instance."""
+
+    # Construct mock return object.
+    json_body = '{"ok": false}'
+
+    resp = _HttpResponse(404, json_body)
+
+    # Patch
+    with patch.object(service, '_make_request', return_value=resp) as mock_func:
+      with self.assertRaises(RuntimeError) as cm:
+        service.get_vm(self.vmpooler_url, self.template_name, self.auth_token)
+
+        excep = cm.exception
+
+        self.assertEqual(excep.msg,
+                         'Could not retrieve template! Invalid template name provided!')
+
+  @skipIf(SKIP_EVERYTHING, 'Skip if we are creating/modifying tests!')
+  def test03_list_vm(self):
+    """Happy path test to verify retrieving the list of available VM instances."""
+
+    # Init
+    expected_machines = ['fake_machine_1', 'fake_machine_1']
+
+    # Construct mock return object.
+    json_body = '[\n  "{0}",\n  "{1}"\n]'.format(expected_machines[0], expected_machines[1])
+
+    resp = _HttpResponse(200, json_body)
+
+    # Patch
+    with patch.object(service, '_make_request', return_value=resp) as mock_func:
+      self.assertListEqual(service.list_vm(self.vmpooler_url,
+                                           self.auth_token),
+                           expected_machines)
+
+  @skipIf(SKIP_EVERYTHING, 'Skip if we are creating/modifying tests!')
+  def test04_info_vm(self):
+    """Happy path test to verify retrieving info about a VM instance."""
+
+    # Init
+    expected_info = {"template": self.template_name,
+                     "lifetime": 23,
+                     "running": 6.7,
+                     "state": "running",
+                     "domain": "delivery.puppetlabs.net"}
+
+    # Construct mock return object.
+    json_body = """
+      {{
+        "ok": true,
+        "{0}": {{
+          "template": "{1}",
+          "lifetime": {2},
+          "running": {3},
+          "state": "{4}",
+          "domain": "{5}"
+        }}
+      }}""".format(self.hostname,
+                   expected_info["template"],
+                   expected_info["lifetime"],
+                   expected_info["running"],
+                   expected_info["state"],
+                   expected_info["domain"])
```

Comment:
One thing you could do if you want to clean this up a bit is:

```python

json_body = """
      {{
        "ok": true,
        "{}": {{
          "template": "{template}",
          "lifetime": {lifetime},
          "running": {running},
          "state": "{state}",
          "domain": "{domain}"
        }}
      }}""".format(self.hostname, **expected_info)

```

Not important though